### PR TITLE
add DoCommand to xArm driver to support xArm gripper

### PR DESCRIPTION
**Pull request by raybjork**
This work adds a DoCommand to the xArm gripper that allows one to move a gripper if it is attached to the xArm6/xArm7.  I'm not sure where this work should live permanently so marking this PR as a draft for now until we can figure that out.  In the meantime @biotinker and @nfranczak can use and add to it as they test.

### CODE REVIEW

File: components/arm/xarm/xarm_comm.go [hunk 1]
```diff
@@ -492,3 +493,49 @@ func getMaxDiff(from, to []referenceframe.Input) float64 {
...
+	if speed {
+		c.params = append(c.params, 0x00, 0x01)
+	} else {
+		c.params = append(c.params, 0x00, 0x00)
+	}
+	_, err := x.send(ctx, c, true)
+	return err
+}
+
+func (x *xArm) setGripperPosition(ctx context.Context, position uint32) error {
```
Comment (biotinker) [inline components/arm/xarm/xarm_comm.go:527]: Can we also add a getter?

Comment (raybjork) [reply components/arm/xarm/xarm_comm.go:527]: Yeah good idea I will add

**Review by randhid — APPROVED** [review]
mmm, are we adding a gripper model? the DoCommands can live here undocumented. But I wouldn't want to add another builtin or go through the scope review process for that.

**Comment by raybjork** [comment]
> mmm, are we adding a gripper model? the DoCommands can live here undocumented. But I wouldn't want to add another builtin or go through the scope review process for that.

I think we want to transition this arm to a module long term.  In that module we can have two resources, (for the gripper and the arm respectively).  But for now I think it existing within DoCommand makes sense (and we can either choose to merge this into main or keep this branch up to date and work off of it)

**Comment by randhid** [comment]
> > mmm, are we adding a gripper model? the DoCommands can live here undocumented. But I wouldn't want to add another builtin or go through the scope review process for that.
> 
> I think we want to transition this arm to a module long term. In that module we can have two resources, (for the gripper and the arm respectively). But for now I think it existing within DoCommand makes sense (and we can either choose to merge this into main or keep this branch up to date and work off of it)

I don't mind merging it, but if you need the gripper model as well, let's just make a private module with the two resources and abandon this one in rdk, it shouldn't have that many uses, and we can easily migrate it.

File: referenceframe/model.go [hunk 2]
```diff
@@ -189,6 +190,18 @@ func (m *SimpleModel) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m.modelConfig)
 }
 
+func (m *SimpleModel) ModelPieceFrames(inputs []Input) (map[string]*staticFrame, error) {
```
Comment (raybjork) [inline referenceframe/model.go:193]: What is this for?

Comment (biotinker) [reply referenceframe/model.go:193]: This allows us to get the positions of the individual pieces of the model, e.g. a specific joint.

Necessary for load calculation.

**Pull request closed without merging**